extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/NarayanaJtaRecorder.java

```diff
@@ -1,28 +1,43 @@
 package io.quarkus.narayana.jta.runtime;
 
-import java.lang.reflect.Field;
 import java.util.Collections;
-import java.util.Properties;
-
-import org.jboss.logging.Logger;
 
+import com.arjuna.ats.arjuna.common.CoordinatorEnvironmentBean;
+import com.arjuna.ats.arjuna.common.CoreEnvironmentBean;
 import com.arjuna.ats.arjuna.common.CoreEnvironmentBeanException;
+import com.arjuna.ats.arjuna.common.ObjectStoreEnvironmentBean;
+import com.arjuna.ats.arjuna.common.RecoveryEnvironmentBean;
 import com.arjuna.ats.arjuna.common.arjPropertyManager;
 import com.arjuna.ats.arjuna.coordinator.TxControl;
+import com.arjuna.ats.jta.common.JTAEnvironmentBean;
 import com.arjuna.ats.jta.common.jtaPropertyManager;
-import com.arjuna.common.util.propertyservice.PropertiesFactory;
+import com.arjuna.common.internal.util.propertyservice.BeanPopulator;
 
 import io.quarkus.runtime.annotations.Recorder;
 
 @Recorder
 public class NarayanaJtaRecorder {
 
-    private static Properties defaultProperties;
+    private static final CoordinatorEnvironmentBean coordinatorEnvironmentBean = new CoordinatorEnvironmentBean();
+    private static final RecoveryEnvironmentBean recoveryEnvironmentBean = new RecoveryEnvironmentBean();
+    private static final CoreEnvironmentBean coreEnvironmentBean = new CoreEnvironmentBean();
+    private static final JTAEnvironmentBean jtaEnvironmentBean = new JTAEnvironmentBean();
+    private static final ObjectStoreEnvironmentBean objectStoreEnvironmentBean = new ObjectStoreEnvironmentBean();
```

Comment:
What's the value of making them static final here instead of initializing them below? It would be nice if could initialize them only if needed btw.

Reply:
I agree it is not ready. I wanted to get something ready for discussion.